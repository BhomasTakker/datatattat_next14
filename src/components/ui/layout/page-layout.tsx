import { MainFooter } from "@/components/footer/main-footer";
import styles from "./page-layout.module.scss";
import { geistMono, geistSans } from "@/fonts";
import { initialiseServices } from "@/lib/services/intialise-services";
import { Providers } from "@/components/providers/Providers";
import "../../../scss/globals.scss";

export const PageLayout = ({ children }: { children: React.ReactNode }) => {
	initialiseServices();
	return (
		<html lang="en">
			<Providers>
				<body
					className={`${geistSans.variable} ${geistMono.variable} ${styles.root}`}
				>
					{children}
					<MainFooter />
				</body>
			</Providers>
		</html>
	);
};
