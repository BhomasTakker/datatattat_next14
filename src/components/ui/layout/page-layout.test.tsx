import React from "react";
import { render, screen } from "@testing-library/react";
import { PageLayout } from "./page-layout";
import "@testing-library/jest-dom";

// Mock dependencies
jest.mock("../../../components/footer/main-footer", () => ({
	MainFooter: () => <footer data-testid="main-footer" />,
}));
jest.mock("../../../components/providers/Providers", () => ({
	Providers: ({ children }: { children: React.ReactNode }) => <>{children}</>,
}));
jest.mock("@vercel/analytics/next", () => ({
	Analytics: () => <div data-testid="analytics" />,
}));
jest.mock("@vercel/speed-insights/next", () => ({
	SpeedInsights: () => <div data-testid="speed-insights" />,
}));
jest.mock("../../../lib/services/intialise-services", () => ({
	initialiseServices: jest.fn().mockResolvedValue(undefined),
}));
jest.mock("../../../fonts", () => ({
	geistMono: { variable: "geistMono-variable" },
	geistSans: { variable: "geistSans-variable" },
}));
jest.mock("./page-layout.module.scss", () => ({
	root: "page-layout-root",
}));

describe("PageLayout", () => {
	it("renders children and layout elements", async () => {
		const TestChild = () => <div data-testid="test-child">Hello</div>;

		const { container } = render(await PageLayout({ children: <TestChild /> }));

		// Render returns a React element, so we need to render it

		expect(screen.getByTestId("test-child")).toBeInTheDocument();
		expect(screen.getByTestId("main-footer")).toBeInTheDocument();
		expect(screen.getByTestId("analytics")).toBeInTheDocument();
		expect(screen.getByTestId("speed-insights")).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it("applies correct class names to body", async () => {
		const { container } = render(await PageLayout({ children: <div /> }));
		const geistSans = container.getElementsByClassName("geistSans-variable");
		expect(geistSans).toHaveLength(1);
		const geistMono = container.getElementsByClassName("geistMono-variable");
		expect(geistMono).toHaveLength(1);
		const page = container.getElementsByClassName("page-layout-root");
		expect(geistSans).toHaveLength(1);
	});

	describe("Snaps", () => {
		it("renders PageLayout unchanged", async () => {
			const { container } = render(await PageLayout({ children: <div /> }));
			expect(container).toMatchSnapshot();
		});
	});
});
