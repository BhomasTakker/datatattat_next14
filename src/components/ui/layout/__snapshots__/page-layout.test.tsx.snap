// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PageLayout Snaps renders PageLayout unchanged 1`] = `
<div>
  <div
    class="geistSans-variable geistMono-variable page-layout-root"
  >
    <div />
    <footer
      data-testid="main-footer"
    />
    <div
      data-testid="analytics"
    />
    <div
      data-testid="speed-insights"
    />
  </div>
</div>
`;

exports[`PageLayout renders children and layout elements 1`] = `
<div>
  <div
    class="geistSans-variable geistMono-variable page-layout-root"
  >
    <div
      data-testid="test-child"
    >
      Hello
    </div>
    <footer
      data-testid="main-footer"
    />
    <div
      data-testid="analytics"
    />
    <div
      data-testid="speed-insights"
    />
  </div>
</div>
`;
